api/queries_repo.go

```diff
@@ -1011,21 +1063,75 @@ func RepoProjects(client *Client, repo ghrepo.Interface) ([]RepoProject, error)
 	return projects, nil
 }
 
+// RepoProjectsV2 fetches all open projectsV2 for a repository
+func RepoProjectsV2(client *Client, repo ghrepo.Interface) ([]RepoProjectV2, error) {
+	type responseData struct {
+		Repository struct {
+			ProjectsV2 struct {
+				Nodes    []RepoProjectV2
+				PageInfo struct {
+					HasNextPage bool
+					EndCursor   string
+				}
+			} `graphql:"projectsV2(first: 100, orderBy: {field: TITLE, direction: ASC}, after: $endCursor)"`
+		} `graphql:"repository(owner: $owner, name: $name)"`
+	}
+
+	variables := map[string]interface{}{
+		"owner":     githubv4.String(repo.RepoOwner()),
+		"name":      githubv4.String(repo.RepoName()),
+		"endCursor": (*githubv4.String)(nil),
+	}
+
+	var projectsV2 []RepoProjectV2
+	for {
+		var query responseData
+		err := client.Query(repo.RepoHost(), "RepositoryProjectV2List", &query, variables)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, p := range query.Repository.ProjectsV2.Nodes {
+			if !p.Closed {
+				projectsV2 = append(projectsV2, p)
+			}
+		}
+
+		if !query.Repository.ProjectsV2.PageInfo.HasNextPage {
+			break
+		}
+		variables["endCursor"] = githubv4.String(query.Repository.ProjectsV2.PageInfo.EndCursor)
+	}
+
+	return projectsV2, nil
+}
+
 // RepoAndOrgProjects fetches all open projects for a repository and its org
-func RepoAndOrgProjects(client *Client, repo ghrepo.Interface) ([]RepoProject, error) {
+func RepoAndOrgProjects(client *Client, repo ghrepo.Interface) ([]RepoProject, []RepoProjectV2, error) {
 	projects, err := RepoProjects(client, repo)
 	if err != nil {
-		return projects, fmt.Errorf("error fetching projects: %w", err)
+		return projects, nil, fmt.Errorf("error fetching projects: %w", err)
 	}
 
 	orgProjects, err := OrganizationProjects(client, repo)
 	// TODO: better detection of non-org repos
 	if err != nil && !strings.Contains(err.Error(), "Could not resolve to an Organization") {
-		return projects, fmt.Errorf("error fetching organization projects: %w", err)
+		return projects, nil, fmt.Errorf("error fetching organization projects: %w", err)
 	}
 	projects = append(projects, orgProjects...)
 
-	return projects, nil
+	projectsV2, err := RepoProjectsV2(client, repo)
```

Comment:
:question: Question from my side. This would mean that users would require to have `read:project` and `read:org` permissions which earlier was not the case. Is it fine to do it this way or should we look up these projects lazily only when there are potential project V2 references?